pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -605,11 +622,19 @@ public boolean commitSegmentMetadata(String rawTableName, final String committin
     final long now = System.currentTimeMillis();
     final String realtimeTableName = TableNameBuilder.REALTIME.tableNameWithType(rawTableName);
 
+    Stat stat = new Stat();
     final LLCRealtimeSegmentZKMetadata oldSegMetadata = getRealtimeSegmentZKMetadata(realtimeTableName,
-        committingSegmentNameStr);
+        committingSegmentNameStr, stat);
     final LLCSegmentName oldSegmentName = new LLCSegmentName(committingSegmentNameStr);
     final int partitionId = oldSegmentName.getPartitionId();
     final int oldSeqNum = oldSegmentName.getSequenceNumber();
+
+    if (oldSegMetadata.getStatus() == CommonConstants.Segment.Realtime.Status.DONE) {
```

Comment:
can u please change the comparison to be NOT IN_PROGRESS?  I feel more comfortable checking for the exact state we should be in, before we continue.